Element: Refactor element package to be React Native friendly

gziolo opened the pull request:
## Description

This PR refactors `@wordpress/element` package to follow the structure we use in other packages. In particular, I broke down the code to be located in the following folders:
* `api` - all the api methods from React and React DOM (including `Fragment` and `StrictMode` from React which might be considered components)
* `components` - our custom components

This enforced also moving tests to new files, as well as updating all local references.

This change will help to maintain a version of the package which works for both web and React Native:
* It gives more granular control over what we override for RN.
* It removes the highly undocumented `react-native` entry point and uses a more robust solution based on the `main` entry point.

#### NPM package

This will temporarily break npm package if we publish it to npm, similar to what we observed in other packages using `export * from ...` because Babel has this bug which adds:
```js
import _Object$defineProperty from 'babel-runtime/core-js/object/define-property';
import _Object$keys from 'babel-runtime/core-js/object/keys';
```

## How has this been tested?
`npm run dev`
`npm run build`
`npm test`
`npm run test-e2e`

## Types of changes
Refactoring only, there is no change in logic.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

gziolo committed:
Element: Refactor element package to be React Native friendly

gziolo committed:
Element: Expose only RawHTML component for React Native version

aduth reviewed (changes requested):
Honestly, I'd rather we just split the files out within the same top-level folder. I don't want to encourage people to think it's fine to add components to this module. It should be an exceedingly rare use case.

gziolo commented:
> Honestly, I'd rather we just split the files out within the same top-level folder. I don't want to encourage people to think it's fine to add components to this module. It should be an exceedingly rare use case.

I will try to open another PR and follow your recommendation. We should have also less code to move around since we extracted `@wordpress/compose` package.
